src/articles/index.tsx

```diff
@@ -15,25 +16,94 @@ import { Header } from '../misc/header';
 
 export function Articles(_: unknown, renderer: RendererLike<Node>) {
   return <>
-    <Route path='**/unapproved' comp={() =>
-      <ArticleList title='Unapproved Articles'
-        articles={fromPromise(getUnapprovedArticles(authToken()!))}
+    <Route path='**/unapproved' comp={() => {
+      const articles = state<Article[]>([]);
+      const isLoading = state(true);
+
+      const getArticles = (lastId?: string) => {
+        isLoading.set(true);
```

Comment:
this function seems in need of being extracted out. besides, we already have a [wait component](https://github.com/Coding-blog/editor-frontend/blob/main/src/misc/wait.tsx) for handling loading from requests.